### PR TITLE
import/export settings buttons

### PR DESCRIPTION
**Pull request by Abeous**
adds buttons to import and export your chat settings (chat.settings in localstorage)
https://i.imgur.com/y71TovV.gifv

### CODE REVIEW

File: assets/chat/js/transfer.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    ChatStore.write('chat.settings', newSettings)
+
+    MessageBuilder.broadcast('Settings imported! Reload chat to take effect.').into(chat)
+  }
+}
+
+function getStorage() {
+  var storage = ChatStore.readRAW('chat.settings')
+  download('settings.txt', storage)
```
Comment (jbpratt78) [inline assets/chat/js/transfer.js]: Is this supposed to be `.json`?

Comment (Abeous) [reply assets/chat/js/transfer.js]: no, but i wasnt aware that the format it is in is valid JSON. i'll commit with a change to json

File: assets/chat/js/transfer.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import ChatStore from './store'
+import {MessageBuilder} from './messages'
+
+function download(filename, text) {
+  var element = document.createElement('a');
+  element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
```
Comment (jbpratt78) [inline assets/chat/js/transfer.js]: I think the `data:text/plain` should be changed to `data:application/json`

File: assets/chat/css/style.scss [hunk 3]
```diff
@@ -1565,3 +1565,25 @@ button.btn {
     background-color: $color-surface-light1;
     border-color: $color-surface-light3;
 }
+
+#importsettingsBtn {
+    display: none;
+}
+
+.importexport {
```
Comment (x-Xymos) [inline assets/chat/css/style.scss]: Shouldn't these any other other variables with this name use kebab case ? (.import-export)
```
.importexport {
.importexport:hover {
```

Comment (Abeous) [reply assets/chat/css/style.scss]: I suppose so, I'm still rusty with my variable naming

**Pull request merged**